addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQCloudHandler.java

```diff
@@ -0,0 +1,277 @@
+     */
+    private void publishBedStatusUpdates() {
+
+        FamilyStatus status = cloud.getFamilyStatus();
+        for (BedStatus bedStatus : status.getBeds()) {
+
+            bedStatusListeners.stream().forEach(l -> l.onBedStateChanged(cloud, bedStatus));
+        }
+    }
+
+    /**
+     * Register the given listener to receive bed status updates.
+     *
+     * @param listener the listener to register
+     */
+    public void registerBedStatusListener(final BedStatusListener listener) {
+
+        if (listener == null) {
+            return;
+        }
+
+        bedStatusListeners.add(listener);
+        publishBedStatusUpdates();
+        updateListenerManagement();
+    }
+
+    /**
+     * Unregister the given listener from further bed status updates.
+     *
+     * @param listener the listener to unregister
+     * @return <code>true</code> if listener was previously registered and is now unregistered; <code>false</code>
+     *         otherwise
+     */
+    public boolean unregisterBedStatusListener(final BedStatusListener listener) {
+
+        boolean result = bedStatusListeners.remove(listener);
+        if (result) {
+            updateListenerManagement();
+        }
+
+        return result;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // all channels are read-only
+    }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+
+        Collection<ConfigStatusMessage> configStatusMessages = new ArrayList<>();
+
+        SleepIQCloudConfiguration config = getConfigAs(SleepIQCloudConfiguration.class);
+        String username = config.username;
+        String password = config.password;
+
+        if (StringUtils.isEmpty(username)) {
+            configStatusMessages.add(ConfigStatusMessage.Builder.error(USERNAME)
+                    .withMessageKeySuffix(SleepIQConfigStatusMessage.USERNAME_MISSING.getMessageKey())
+                    .withArguments(USERNAME).build());
+        }
+
+        if (StringUtils.isEmpty(password)) {
+            configStatusMessages.add(ConfigStatusMessage.Builder.error(PASSWORD)
+                    .withMessageKeySuffix(SleepIQConfigStatusMessage.PASSWORD_MISSING.getMessageKey())
+                    .withArguments(PASSWORD).build());
+        }
+
+        return configStatusMessages;
+    }
+
+    /**
+     * Get a list of all beds registered to the cloud service account.
+     *
+     * @return the list of beds (never <code>null</code>)
+     */
+    public List<Bed> getBeds() {
+
+        return cloud.getBeds();
+    }
+
+    /**
+     * Get the {@link Bed} corresponding to the given identifier.
+     *
+     * @param bedId the bed identifier
+     * @return the identified {@link Bed} or <code>null</code> if no such bed exists
+     */
+    public Bed getBed(final String bedId) {
+
+        for (Bed bed : getBeds()) {
+
```

Comment:
I will remove all of the empty lines after opening braces. I am used to the style of putting opening braces on a new line, so it appears I added these new lines somewhat subconsciously.